sklearn/ensemble/tests/test_gradient_boosting_loss_functions.py

```diff
@@ -320,3 +320,18 @@ def test_lad_equals_quantiles(seed, alpha):
         y_true, raw_predictions, sample_weight=weights, alpha=alpha
     )
     assert pbl_weighted_loss == approx(ql_weighted_loss)
+
+
+def test_exponential_loss():
```

Comment:
I would prefer to compare the gradients to numerical differentiation for all losses, but this certainly is a regression test for this bug.

Reply:
We can have add these tests for the common losses I hope :)

Reply:
That the point: they are already there :wink: